internal/notifier/google_pubsub_topic_test.go

```diff
@@ -0,0 +1,139 @@
+	return tt.expectedCloseErr
+}
+
+func TestGooglePubSubTopic(t *testing.T) {
+	for name, tt := range map[string]*googlePubSubTestCase{
+		"invalid topic names return error": {
+			topicName:        "invalid-topic-name",
+			expectedParseErr: errors.New("topic name doesn't match 'projects/<projectID>/topics/<topicID>': 'invalid-topic-name'"),
+		},
+		"project and topic ids are parsed from the topic name": {
+			topicName:         "projects/parsedProjectID/topics/parsedTopicID",
+			expectedProjectID: "parsedProjectID",
+			expectedTopicID:   "parsedTopicID",
+			expectedMessage:   `{"involvedObject":{},"severity":"","timestamp":null,"message":"","reason":"","reportingController":""}`,
+		},
+		"commit status updates are dropped": {
+			topicName:         "projects/projectID/topics/topicID",
+			expectedProjectID: "projectID",
+			expectedTopicID:   "topicID",
+			event: eventv1.Event{
+				Metadata: map[string]string{"commit_status": "update"},
+			},
+		},
+		"events are properly marshaled": {
+			topicName:         "projects/projectID/topics/topicID",
+			expectedProjectID: "projectID",
+			expectedTopicID:   "topicID",
+			event: eventv1.Event{
+				Metadata: map[string]string{"foo": "bar"},
+			},
+			expectedMessage: `{"involvedObject":{},"severity":"","timestamp":null,"message":"","reason":"","metadata":{"foo":"bar"},"reportingController":""}`,
+		},
+		"publish error is relayed": {
+			topicName:         "projects/projectID/topics/topicID",
+			expectedProjectID: "projectID",
+			expectedTopicID:   "topicID",
+			expectedMessage:   `{"involvedObject":{},"severity":"","timestamp":null,"message":"","reason":"","reportingController":""}`,
+			publishErr:        errors.New("publish error"),
+			expectedPostErr:   fmt.Errorf("error publishing event to topic projects/projectID/topics/topicID: %w", errors.New("publish error")),
+		},
+		"close error is relayed": {
+			topicName:         "projects/projectID/topics/topicID",
+			expectedProjectID: "projectID",
+			expectedTopicID:   "topicID",
+			expectedMessage:   `{"involvedObject":{},"severity":"","timestamp":null,"message":"","reason":"","reportingController":""}`,
+			expectedCloseErr:  errors.New("close error"),
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			projectID, topicID, err := ParseGooglePubSubTopic(tt.topicName)
+			g.Expect(projectID).To(Equal(tt.expectedProjectID))
+			g.Expect(topicID).To(Equal(tt.expectedTopicID))
+			if tt.expectedParseErr == nil {
+				g.Expect(err).To(BeNil())
+			} else {
+				g.Expect(err).To(Equal(tt.expectedParseErr))
+			}
+
+			if err != nil {
+				return
+			}
+
+			topic := &GooglePubSubTopic{
+				client:    tt,
+				topicName: tt.topicName,
+				topicID:   topicID,
+			}
+
+			err = topic.Post(context.Background(), tt.event)
+			if tt.expectedPostErr == nil {
+				g.Expect(err).To(BeNil())
+			} else {
+				g.Expect(err).To(Equal(tt.expectedPostErr))
+			}
+
+			err = topic.Close()
```

Comment:
Same here, this should be tested in a separate test case (i.e. `Test...` func) that can be run individually.